CMakeLists.txt

```diff
@@ -223,10 +225,10 @@ else()
 	check_and_add_flag(VISIBILITY_INLINES_HIDDEN -fvisibility-inlines-hidden)
 	check_and_add_flag(VISIBILITY_HIDDEN -fvisibility=hidden)
 
-	check_c_compiler_flag(-fomit-frame-pointer FLAG_C_FOMIT_FRAME_POINTER)
-	if(FLAG_C_FOMIT_FRAME_POINTER)
-		add_compile_options($<$<CONFIG:Release>:-fomit-frame-pointer>)
-	endif()
+	check_and_add_flag(FOMIT_FRAME_POINTER -fomit-frame-pointer RELEASE)
+
+	add_compile_definitions(_DEBUG DEBUG)
+	check_and_add_flag(GGDB -ggdb DEBUG)
```

Comment:
Another possible improvement (unrelated to this PR) I thought of is, for most of these flag checks the variable is 1) unused and 2) derived pretty directly from the actual flag. It could be nice to get rid of specifying the variable, so you'd have something like

```cmake
check_and_add_flags(-pg)
if(NOT FLAG_C_-pg)
    message(FATAL_ERROR "Ya done goofed")
    # ...
```